cloudinit/tests/test_url_helper.py

```diff
@@ -71,6 +75,25 @@ def test_read_file_or_url_str_from_url(self):
         self.assertEqual(result.contents, data)
         self.assertEqual(str(result), data.decode('utf-8'))
 
+    @httpretty.activate
+    def test_read_file_or_url_str_from_url_redacting_headers_from_logs(self):
+        """Headers are redacted from logs but unredacted in requests."""
+        url = 'http://hostname/path'
+
+        # httpretty 0.9.6 Xenial doens't support httpretty.latest_requests
```

Comment:
Can we comment here that this is how we verify that all headers are submitted vs. what gets logged.